lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+        if self._should_merge(data, resource, key=SPKeys.BOOT_MODE):
+            merged_data = self._merge_dict(merged_data, resource, data, key=SPKeys.BOOT_MODE)
+        return merged_data
+
+    def _merge_connections(self, merged_data, resource, data):
+        if self._should_merge(data, resource, key=SPKeys.CONNECTIONS):
+            existing_connections = resource[SPKeys.CONNECTIONS]
+            params_connections = data[SPKeys.CONNECTIONS]
+            merged_data[SPKeys.CONNECTIONS] = ResourceMerger.merge_list_by_key(existing_connections,
+                                                                               params_connections,
+                                                                               key=SPKeys.ID)
+
+            merged_data = self._merge_connections_boot(merged_data, resource)
+        return merged_data
+
+    def _merge_connections_boot(self, merged_data, resource):
+        existing_connection_map = {}
+        for x in resource[SPKeys.CONNECTIONS]:
+            existing_connection_map[x[SPKeys.ID]] = x.copy()
+        for merged_connection in merged_data[SPKeys.CONNECTIONS]:
+            conn_id = merged_connection[SPKeys.ID]
+            existing_conn_has_boot = conn_id in existing_connection_map and SPKeys.BOOT in existing_connection_map[
+                conn_id]
+            if existing_conn_has_boot and SPKeys.BOOT in merged_connection:
+                current_connection = existing_connection_map[conn_id]
+                boot_settings_merged = deepcopy(current_connection[SPKeys.BOOT])
+                boot_settings_merged.update(merged_connection[SPKeys.BOOT])
+                merged_connection[SPKeys.BOOT] = boot_settings_merged
+        return merged_data
+
+    def _merge_san_storage(self, merged_data, data, resource):
+        if self._removed_data(data, resource, key=SPKeys.SAN):
+            merged_data[SPKeys.SAN] = dict(volumeAttachments=[], manageSanStorage=False)
+        elif self._should_merge(data, resource, key=SPKeys.SAN):
+            merged_data = self._merge_dict(merged_data, resource, data, key=SPKeys.SAN)
+
+            merged_data = self._merge_san_volumes(merged_data, resource, data)
+        return merged_data
+
+    def _merge_san_volumes(self, merged_data, resource, data):
+        if self._should_merge(data[SPKeys.SAN], resource[SPKeys.SAN], key=SPKeys.VOLUMES):
+            existing_volumes = resource[SPKeys.SAN][SPKeys.VOLUMES]
+            params_volumes = data[SPKeys.SAN][SPKeys.VOLUMES]
+            merged_volumes = ResourceMerger.merge_list_by_key(existing_volumes, params_volumes, key=SPKeys.ID)
+            merged_data[SPKeys.SAN][SPKeys.VOLUMES] = merged_volumes
+
+            merged_data = self._merge_san_storage_paths(merged_data, resource)
+        return merged_data
+
+    def _merge_san_storage_paths(self, merged_data, resource):
+        existing_volumes_map = collections.OrderedDict([(i[SPKeys.ID], i) for i in resource[
```

Comment:
It doesn't look like you need an OrderedDict here.  A regular dict will do (and be faster and take up less memory).